src/hal-resource.ts

```diff
@@ -147,4 +152,23 @@ export class HalResource implements IHalResource {
 
       return json;
     }
+
+    private serializeProperty(prop, serializer: IJSONSerializer, arrayItem?: any) {

```

Comment:
I don't think this is a correct place for this method.
There are a `hal-json-serializer` file to serialize objects.

Reply:
This is just an extension to `serialize(...)` . It is just the part that I tool from `serialize(...)` to make it callable recursively (lines 143-147). And I added the ability to process arrays. Maybe the name should be something else.

Line 143 of the previous version corresponds to line 161. I used some helper methods of lodash to improve readability.
Old line 144 equals new line 162.
So is old line 146 to new line 169. They are the same.
In between is the ability to read arrays (and the recursive call).